.buildkite/pipeline-nightly-e2e-tests.yml

```diff
@@ -0,0 +1,348 @@
+          image: "family/elastic-buildkite-agent-ubuntu-2004-lts"
+          machineType: "n1-standard-16"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/kind-k8s-versions kindest/node:v1.22.9@sha256:8135260b959dfe320206eb36b3aeda9cffcb262f4b44cda6b33f7bb73f453105 0.14.0 ipv6
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make -C .ci TARGET="run-deployer e2e-run" ci
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
+
+      - label: "kind 1.23 ipv4"
+        agents:
+          provider: "gcp"
+          image: "family/elastic-buildkite-agent-ubuntu-2004-lts"
+          machineType: "n1-standard-16"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/kind-k8s-versions kindest/node:v1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae 0.14.0 ipv4
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make -C .ci TARGET="run-deployer e2e-run" ci
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
+
+      - label: "kind 1.23 ipv6"
+        agents:
+          provider: "gcp"
+          image: "family/elastic-buildkite-agent-ubuntu-2004-lts"
+          machineType: "n1-standard-16"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/kind-k8s-versions kindest/node:v1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae 0.14.0 ipv6
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make -C .ci TARGET="run-deployer e2e-run" ci
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
+
+      - label: "kind 1.24 ipv4"
+        agents:
+          provider: "gcp"
+          image: "family/elastic-buildkite-agent-ubuntu-2004-lts"
+          machineType: "n1-standard-16"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/kind-k8s-versions kindest/node:v1.24.1@sha256:11276ce4763d5600379c0104c4cb51bf7420c8c02937708e958ad68d08e0910c 0.14.0 ipv4
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make -C .ci TARGET="run-deployer e2e-run" ci
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
+
+      - label: "kind 1.24 ipv6"
+        agents:
+          provider: "gcp"
+          image: "family/elastic-buildkite-agent-ubuntu-2004-lts"
+          machineType: "n1-standard-16"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/kind-k8s-versions kindest/node:v1.24.1@sha256:11276ce4763d5600379c0104c4cb51bf7420c8c02937708e958ad68d08e0910c 0.14.0 ipv6
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make -C .ci TARGET="run-deployer e2e-run" ci
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
+
+  - group: "ocp"
+    steps:
+
+      - label: "ocp 4.7"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/ocp 4.7.49
```

Comment:
I remember you explained the need for this but I still don't fully understand/remember. Can you explain why we have to call `setenvconfig` twice and would not the second call overwrite everything the first one produced?

Reply:
This is a twisted way to pass the `OPERATOR_IMAGE` to use for the e2e-tests.
The first call `.ci/setenvconfig e2e/main` (same as `.ci/setenvconfig pr`) sets:
```
IMG_SUFFIX=-ci
REGISTRY_NAMESPACE=eck-snapshots
```

Then `.ci/setenvconfig` sets `JKS_PARAM_OPERATOR_IMAGE` by calling `make print-operator-image` which uses the precedent variables to assemble the operator image name:

```
  if [[ "$environment" =~ e2e/* ]]; then
    JKS_PARAM_OPERATOR_IMAGE=$(make print-operator-image)
  fi

# OPERATOR_IMAGE is resolved in this way:
# OPERATOR_IMAGE ?= docker.elastic.co/$(REGISTRY_NAMESPACE)/eck-operator$(IMG_SUFFIX):$$(cat VERSION)-$$(git rev-parse --short=8 --verify HEAD)
```

This is a variation of what we're doing today:

https://github.com/elastic/cloud-on-k8s/blob/d8aa26be4c25783c4e98a189476c3d1779f05d23/.ci/pipelines/build.Jenkinsfile#L79-L83


----

As an alternative, we could use the [build meta-data](https://buildkite.com/docs/pipelines/build-meta-data) to pass the operator image between the steps:
```yaml
      - label: ":docker: operator container image"
        commands:
          - .ci/setenvconfig pr
          - .buildkite/scripts/common/get-test-artifacts.sh
          - make operator-buildah
          - buildkite-agent meta-data set "operator-image" "$(make print-operator-image)"
...
      - label: "aks"
        commands:
          - JKS_PARAM_OPERATOR_IMAGE=$(buildkite-agent meta-data get "operator-image") .ci/setenvconfig e2e/aks
```